iib/workers/tasks/greenwave.py

```diff
@@ -0,0 +1,89 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import json
+import logging
+
+import requests
+
+from iib.exceptions import IIBError
+from iib.workers.config import get_worker_config
+from iib.workers.tasks.utils import get_image_labels
+
+log = logging.getLogger(__name__)
+
+
+def gate_bundles(bundles):
+    """
+    Check if all bundle images have passed gating tests in the CVP pipeline.
+
+    This method queries Greenwave for the following policy:
+    https://gitlab.cee.redhat.com/ansible-playbooks/greenwave-playbooks/blob/master/files/prod/redhat.yaml#L313
+
+    :param list bundles: a list of strings representing the pull specifications of the bundles to
+        be gated.
+    :raises IIBError: if any of the bundles fail the gating checks or IIB fails to get a
+        response from Greenwave.
+    """
+    conf = get_worker_config()
+    _verify_greenwave_config(conf)
```

Comment:
The reason I do it here is because this addition of Gating to IIB is temporary (until ET-Greenwave integration is done). It would be easier to clean it up later.

Reply:
Since it is now optional, I have still kept it there since we need that check only when Gating is enabled for the queue.